src/aotcompile.cpp

```diff
@@ -1951,13 +1944,10 @@ extern "C" JL_DLLEXPORT_CODEGEN jl_code_info_t *jl_gdbdumpcode(jl_method_instanc
         src = (jl_code_info_t*)jl_atomic_load_relaxed(&codeinst->inferred);
         if ((jl_value_t*)src != jl_nothing && !jl_is_code_info(src) && jl_is_method(mi->def.method)) {
             JL_GC_PUSH2(&codeinst, &src);
-            src = jl_uncompress_ir(mi->def.method, codeinst, (jl_value_t*)src);
+            src = jl_uncompress_ir(mi->def.method, (jl_value_t*)src);
             JL_GC_POP();
         }
     }
-    if (!src || (jl_value_t*)src == jl_nothing) {
-        src = jl_type_infer(mi, world, 0);
```

Comment:
While this is just a debug helper for convenience from gdb, it would be nice to be able to keep the ability here that it returns an uncompressed source object, so that this function continues to do the full complement of code operations (code_native, code_ir, code_typed). I think that just needs the ABI_SOURCE argument?